### PR TITLE
Nc/callbacks docs

### PR DESCRIPTION
**Pull request by nfcampos**

### CODE REVIEW

File: docs/docs/modules/chains/index.mdx [hunk 1]
```diff
@@ -17,3 +18,23 @@ Using a language model in isolation is fine for some applications, but it is oft
...
+
+To implement your own custom chain you can subclass `BaseChain` and implement the following methods:
+
+import SubclassInterface from "@examples/chains/advanced_subclass.ts";
+
+<CodeBlock language="typescript">{SubclassInterface}</CodeBlock>
+
+### Subclassing `BaseChain`
+
+The `_call` method is the main method of a chain. It takes a record of inputs and returns a record of outputs. The inputs received should conform the `inputKeys` array, and the outputs returned should conform to the `outputKeys` array.
```
Comment (agola11) [inline docs/docs/modules/chains/index.mdx:32]: instead of "main method of a chain," how about "main method custom chains must implement"

**Review by agola11 — APPROVED** [review]
looks good, I like the consolidation of chaininputs too

File: docs/docs/modules/chains/index.mdx (hunk 1 above)
Comment (nfcampos) [reply docs/docs/modules/chains/index.mdx:32]: Agreed!

**Comment by nfcampos** [comment]
yea the consolidation was just to make it easier to create a custom chain (and actually fixed a few silent bugs of chains that were ignoring `callbacks` arg

File: docs/docs/production/callbacks/index.mdx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+import DocCardList from "@theme/DocCardList";
+
+<DocCardList />
+
+## How to use callbacks
+
+The `callbacks` argument is available on most objects throughout the API (Chains, Models, Tools, Agents, etc.) in two different places:
+
+- **Constructor callbacks**: defined in the constructor, eg. `new LLMChain({ callbacks: [handler] })`, which will be used for all calls made on that object, and will be scoped to that object only, ie doen't affect any child object, eg. if you pass a handler to the LLMChain constructor, it will not be used by the Model attached to that chain.
```
Comment (agola11) [inline docs/docs/production/callbacks/index.mdx:19]: I think this is actually swapped. Constructor callbacks are scoped to that particular object and not passed through / inherited. Request callbacks are piped through all sub objects

**Pull request merged**